crates/flux-fhir-analysis/src/annot_check.rs

```diff
@@ -29,60 +31,68 @@ pub fn check_fn_sig(
     if fn_sig.lifted {
         return Ok(());
     }
-    let self_ty = lift::lift_self_ty(tcx, sess, owner_id)?;
-    let expected_fn_sig = &lift::lift_fn(tcx, sess, owner_id)?.1.fn_sig;
-    Zipper::new(sess, wfckresults, self_ty.as_ref()).zip_fn_sig(fn_sig, expected_fn_sig)
+    let self_ty = lift::lift_self_ty(genv.tcx, sess, owner_id)?;
+    let expected_fn_sig = &lift::lift_fn(genv.tcx, sess, owner_id)?.1.fn_sig;
+    Zipper::new(sess, wfckresults, self_ty.as_ref()).zip_fn_sig(genv, fn_sig, expected_fn_sig)
 }
 
 pub fn check_alias(
-    tcx: TyCtxt,
+    genv: &GlobalEnv,
+    // tcx: TyCtxt,
     sess: &FluxSession,
     wfckresults: &mut WfckResults,
     ty_alias: &fhir::TyAlias,
 ) -> Result<(), ErrorGuaranteed> {
     if ty_alias.lifted {
         return Ok(());
     }
-    let (.., expected_ty_alias) = lift::lift_type_alias(tcx, sess, ty_alias.owner_id)?;
-    Zipper::new(sess, wfckresults, None).zip_ty(&ty_alias.ty, &expected_ty_alias.ty)
+    let (.., expected_ty_alias) = lift::lift_type_alias(genv.tcx, sess, ty_alias.owner_id)?;
+    Zipper::new(sess, wfckresults, None).zip_ty(genv, &ty_alias.ty, &expected_ty_alias.ty)
 }
 
 pub fn check_struct_def(
-    tcx: TyCtxt,
+    genv: &GlobalEnv,
     sess: &FluxSession,
     wfckresults: &mut WfckResults,
     struct_def: &fhir::StructDef,
 ) -> Result<(), ErrorGuaranteed> {
     match &struct_def.kind {
         fhir::StructKind::Transparent { fields } => {
-            let mut liftcx = LiftCtxt::new(tcx, sess, struct_def.owner_id, None);
+            let mut liftcx = LiftCtxt::new(genv.tcx, sess, struct_def.owner_id, None);
             fields.iter().try_for_each_exhaust(|field| {
                 if field.lifted {
                     return Ok(());
                 }
-                let self_ty = lift::lift_self_ty(tcx, sess, struct_def.owner_id)?;
-                Zipper::new(sess, wfckresults, self_ty.as_ref())
-                    .zip_ty(&field.ty, &liftcx.lift_field_def_id(field.def_id)?.ty)
+                let self_ty = lift::lift_self_ty(genv.tcx, sess, struct_def.owner_id)?;
+                Zipper::new(sess, wfckresults, self_ty.as_ref()).zip_ty(
+                    genv,
```

Comment:
Instead of threading down `genv` we could add it to a field in the zipper (instead of `FluxSession`)